.github/workflows/golang-lint.yml

```diff
@@ -0,0 +1,24 @@
+name: golint
+on:
+  push:
+    tags:
+      - v*
+    branches:
+      - master
+  pull_request:
+jobs:
+  golangci-lint:
+    name: runner / golangci-lint
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check out code into the Go module directory
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+      - name: golangci-lint
+        uses: reviewdog/action-golangci-lint@v2
+        with:
+          reviewdog_version: 'v0.13.1'
+          golangci_lint_version: 'v1.43.0'
+          golangci_lint_flags:  '--new-from-rev=HEAD~'
+          level: warning # GitHub Status Check won't become failure with this level.
```

Comment:
I have been using the official action in other repos. It has good integration with GitHub. 
https://github.com/golangci/golangci-lint-action

Reply:
> but I want to add the reviewdog GitHub commiter

May I ask why?

Reply:
sure, it simplifies to see why lint fails, but as I said it is a WIP, will see

Reply:
What I meant is that the official action also shows this.